notes_app/client/notes.js

```diff
@@ -85,15 +85,15 @@
   angular.module('notes').run(function($rootScope, $state, userService, $q) {
     $rootScope.$on('$stateChangeStart',
       function(event, toState, toParams, fromState, fromParams, options) {
-        var user = angular.fromJson(localStorage.user);
+        userService.checkSession().then(function(session){
```

Comment:
you're still trying to load session on every state change. It doesn't look good. You have to do it only when page is refreshed or app is opening in browser